parl/framework/tests/algorithm_test.py

```diff
@@ -0,0 +1,126 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle.fluid as fluid
+import parl.layers as layers
+from parl.framework.model_base import Model
+from parl.framework.algorithm_base import Algorithm
+from copy import deepcopy
+import numpy as np
+import unittest
+import sys
+
+
+class Value(Model):
+    def __init__(self, obs_dim, act_dim):
+        self.obs_dim = obs_dim
+        self.act_dim = act_dim
+
+        self.fc1 = layers.fc(size=256, act='relu')
+        self.fc2 = layers.fc(size=128, act='relu')
+        self.fc3 = layers.fc(size=self.act_dim)
+
+    def value(self, obs):
+        out = self.fc1(obs)
+        out = self.fc2(out)
+        value = self.fc3(out)
+        return value
+
+
+class QLearning(Algorithm):
+    def __init__(self, critic_model):
+        self.critic_model = critic_model
+        self.target_model = deepcopy(critic_model)
+
+    def define_predict(self, obs):
```

Comment:
self.q_target_value = self.target_model.value(obs) can be placed in define_learn (or define_target_predict) function, which make one_program and multi_programs testing more conveniently.

Reply:
Good suggestion. The most important point for unit test is to ensure that the code has covered all the function defined.